File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {

        int n = nums.length;
        k %= n;
        reverseArray(nums, 0 , n-k-1);
        reverseArray(nums, n-k , n-1);
        reverseArray(nums,0 , n-1);


        
    }

    public static void reverseArray(int nums[], int i, int j){
        while(i < j){
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++;
            j--;
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k = k % nums.size();
        reverse(nums,0,nums.size()-k-1);
        reverse(nums,nums.size()-k, nums.size()-1);
        reverse(nums, 0, nums.size()-1);


        
    }
    void reverse(vector<int>& nums, int i , int j){
        while(i < j){
            swap(nums[i], nums[j]);
            i++;
            j--;
        }


    }
};